lib/src/ardrive_http.dart

```diff
@@ -87,6 +92,61 @@ class ArDriveHTTP {
     return get(url: url, responseType: ResponseType.bytes);
   }
 
+  Future<ArDriveHTTPResponse> getAsByteStream(String url, {Completer<String>? cancelWithReason}) async {
+    return kIsWeb 
+      ? _getAsByteStreamWeb(url, cancelWithReason: cancelWithReason)
+      : _getAsByteStreamIO(url, cancelWithReason: cancelWithReason);
```

Comment:
We can use only one method, as the only change is the `client` and `cancellationToken` (only on io). It will be easier to maintain.

Reply:
I tried this: https://github.com/elliotsayes/ardrive_http/compare/http-client-stream...elliotsayes:ardrive_http:http-client-stream-combine
But `test-vm` gives errors like this: 
```
  type 'IOClient' is not a subtype of type 'Client' in type cast
  package:ardrive_http/src/ardrive_http.dart 100:49  ArDriveHTTP.getAsByteStream
  test/ardrive_http_test.dart 69:40                  main.<fn>.<fn>.<fn>
```
I think this is because `cancellation_token_http` is a **fork** of `http` rather than just a subclass, and therefore has a different class hierarchy to `fetch_client`

Reply:
If you had a different way in mind just let me know. I guess I can keep the client, response as `Object`/`dynamic` but I don't think this would really be an improvement